tools/setup_services.py

```diff
@@ -441,15 +458,17 @@ def update_or_clone_plugin(
         return True
 
     if version_tag:
-        log(f"Updating {plugin_name} to version {version_tag}")
         return update_or_clone_plugin_by_tag(url, version_tag, plugin_name, plugin_path)
     elif branch and sha:
-        log(f"Updating {plugin_name} to branch {branch} at SHA {sha}")
         return update_or_clone_plugin_by_branch(
             url, branch, sha, plugin_name, plugin_path
         )
-    else:
+    elif os.path.isdir(plugin_path):
         return True
+    # TODO: if it exists but it's an uninitialized git submodule
```

Comment:
maybe let's remove this TODO, feels like submodules could be handled in a separate PR. In general I think we shouldn't try to do EVERYTHING imaginable in this one, and better merge a first version we are happy with.